bfvmm/src/vmcs/src/vmcs_intel_x64.cpp

```diff
@@ -496,39 +543,39 @@ vmcs_intel_x64::secondary_processor_based_vm_execution_controls()
 void
 vmcs_intel_x64::vm_exit_controls()
 {
-    auto controls = vm::read(VMCS_VM_EXIT_CONTROLS);
-
-    controls |= VM_EXIT_CONTROL_SAVE_DEBUG_CONTROLS;
-    controls |= VM_EXIT_CONTROL_HOST_ADDRESS_SPACE_SIZE;
-    controls |= VM_EXIT_CONTROL_LOAD_IA32_PERF_GLOBAL_CTRL;
-    controls |= VM_EXIT_CONTROL_ACKNOWLEDGE_INTERRUPT_ON_EXIT;
-    controls |= VM_EXIT_CONTROL_SAVE_IA32_PAT;
-    controls |= VM_EXIT_CONTROL_LOAD_IA32_PAT;
-    controls |= VM_EXIT_CONTROL_SAVE_IA32_EFER;
-    controls |= VM_EXIT_CONTROL_LOAD_IA32_EFER;
-    // controls |= VM_EXIT_CONTROL_SAVE_VMX_PREEMPTION_TIMER_VALUE;
+    auto controls = vmcs::vm_exit_controls::get();
+
+    controls |= vmcs::vm_exit_controls::save_debug_controls::mask;
+    controls |= vmcs::vm_exit_controls::host_address_space_size::mask;
+    controls |= vmcs::vm_exit_controls::load_ia32_perf_global_ctrl::mask;
+    controls |= vmcs::vm_exit_controls::acknowledge_interrupt_on_exit::mask;
+    controls |= vmcs::vm_exit_controls::save_ia32_pat::mask;
+    controls |= vmcs::vm_exit_controls::load_ia32_pat::mask;
+    controls |= vmcs::vm_exit_controls::save_ia32_efer::mask;
+    controls |= vmcs::vm_exit_controls::load_ia32_efer::mask;
+    // controls |= vmcs::vm_exit_controls::save_vmx_preemption_timer_value::mask;
 
     this->filter_unsupported(msrs::ia32_vmx_true_exit_ctls::addr, controls);
 
-    vm::write(VMCS_VM_EXIT_CONTROLS, controls);
+    vmcs::vm_exit_controls::set(controls);
 }
 
 void
 vmcs_intel_x64::vm_entry_controls()
 {
-    auto controls = vm::read(VMCS_VM_ENTRY_CONTROLS);
+    auto controls = vmcs::vm_entry_controls::get();
 
-    controls |= VM_ENTRY_CONTROL_LOAD_DEBUG_CONTROLS;
-    controls |= VM_ENTRY_CONTROL_IA_32E_MODE_GUEST;
-    // controls |= VM_ENTRY_CONTROL_ENTRY_TO_SMM;
-    // controls |= VM_ENTRY_CONTROL_DEACTIVATE_DUAL_MONITOR_TREATMENT;
-    controls |= VM_ENTRY_CONTROL_LOAD_IA32_PERF_GLOBAL_CTRL;
-    controls |= VM_ENTRY_CONTROL_LOAD_IA32_PAT;
-    controls |= VM_ENTRY_CONTROL_LOAD_IA32_EFER;
+    controls |= vmcs::vm_entry_controls::load_debug_controls::mask;
+    controls |= vmcs::vm_entry_controls::ia_32e_mode_guest::mask;
+    // controls |= vmcs::vm_entry_controls::entry_to_smm::mask;
+    // controls |= vmcs::vm_entry_controls::deactivate_dual_monitor_treatment::mask;
+    controls |= vmcs::vm_entry_controls::load_ia32_perf_global_ctrl::mask;
+    controls |= vmcs::vm_entry_controls::load_ia32_pat::mask;
+    controls |= vmcs::vm_entry_controls::load_ia32_efer::mask;
 
     this->filter_unsupported(msrs::ia32_vmx_true_entry_ctls::addr, controls);
```

Comment:
I'm going to ypdate all of these in the next PR with the changes for the secondary controls.